### PR TITLE
[ETFE-3504] - Added EPC in error handling

### PR DESCRIPTION
**Pull request by LewisMitchell**

### CODE REVIEW

**Comment by LewisMitchell** [comment]
Need to add warning text when in CheckMode

File: app/pages/sections/items/ItemExciseProductCodePage.scala [hunk 1]
```diff
@@ -16,11 +16,39 @@
...
+
+  private val itemExciseProductCodesError: Seq[String] = Seq(
+    itemExciseProductCodeConsignorNotApprovedToSendError,
+    itemExciseProductCodeConsigneeNotApprovedToReceiveError,
+    itemExciseProductCodeDestinationNotApprovedToReceiveError,
+    itemExciseProductCodeDispatchPlaceNotAllowed
+  )
+
+  private def isExciseProductCodeAtIndex: MovementSubmissionFailure => Boolean = submissionFailure =>
+    submissionFailure.errorLocation.exists(_.contains(s"$BODYEADESAD[${idx.position + 1}]")) &&
```
Comment (machinesleet) [inline app/pages/sections/items/ItemExciseProductCodePage.scala:40]: Not sure if it's over-abstraction, but we've got 5 places in the repo that do this kind of check. Maybe a little helper method somewhere? Also wondering if the BODYEADESAD string could be a constant, but see we've already got a Pascal case one in there. Would it be worth adding a constant and differentiating between the two?

Comment (LewisMitchell) [reply app/pages/sections/items/ItemExciseProductCodePage.scala]: Done in 3cada8db283b4cf563861e068084de24e44a5b24

File: app/pages/sections/items/ItemExciseProductCodePage.scala [hunk 2]
```diff
@@ -16,11 +16,39 @@
...
+    request.userAnswers.submissionFailures.find(isExciseProductCodeAtIndex)
+
+  override def isMovementSubmissionError(implicit request: DataRequest[_]): Boolean =
+    getMovementSubmissionFailure.exists(!_.hasBeenFixed)
+
+  override def getOriginalAttributeValue(implicit request: DataRequest[_]): Option[String] =
+    getMovementSubmissionFailure.flatMap(_.originalAttributeValue)
+
+  def getSubmissionErrorCodes(isOnAddToList: Boolean)(implicit request: DataRequest[_]): Seq[ErrorCode] =
+    request.userAnswers.submissionFailures.filter(isExciseProductCodeAtIndex).filter(!_.hasBeenFixed).map(error => ErrorCode(error.errorType, idx, isOnAddToList))
```
[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/pages/sections/items/ItemExciseProductCodePage.scala]: Done in 3cada8db283b4cf563861e068084de24e44a5b24

**Review by machinesleet — APPROVED** [review]

**Pull request merged**